tests/unittests/Foundation/NSThreadTests.m

```diff
@@ -42,6 +42,12 @@ - (void)countingloop {
 static const double sExpectedLowThreadPriority = 0;
 static const double sExpectedMidThreadPriority = .5;
 
+// A thread never has a set priority. The system can always change it, in fact it will most likely change the priority.
+// Even still, the thread priority should be close enough to the actual requested value so we need to make sure it's sufficiently close.
+// The largest difference between thread priorities in Windows should be .07 units in difference as Windows pigeon holes priorities into
+// certain values.
+static const double sMarginOfError = .07;
```

Comment:
This is interesting. Can the system both increase and decrease the priority? I'm wondering if there are stronger statements we can make or if this is it. Also  how exactly did you arrive at .07? given that this needs to run on OSX and Windows we want the max of the two's range of movement right?